core/capabilities/ccip/launcher/deployment.go

```diff
@@ -1,142 +1,45 @@
 package launcher
 
 import (
-	"fmt"
+	"golang.org/x/exp/maps"
 
 	cctypes "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
-
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
 	"go.uber.org/multierr"
-
-	ccipreaderpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
 )
 
-// activeCandidateDeployment represents a active-candidate deployment of OCR instances.
-type activeCandidateDeployment struct {
-	// active is the active OCR instance.
-	// active must always be present.
-	active cctypes.CCIPOracle
-
-	// candidate is the candidate OCR instance.
-	// candidate may or may not be present.
-	// candidate must never be present if active is not present.
-	candidate cctypes.CCIPOracle
-}
-
-// ccipDeployment represents active-candidate deployments of both commit and exec
-// OCR instances.
-type ccipDeployment struct {
-	commit activeCandidateDeployment
-	exec   activeCandidateDeployment
-}
-
-// Close shuts down all OCR instances in the deployment.
-func (c *ccipDeployment) Close() error {
-	// we potentially run into this situation when
-	// trying to close an active instance that doesn't exist
-	// this check protects us from nil pointer exception
+type ccipPlugins map[ocrtypes.ConfigDigest]*cctypes.CCIPOracle
 
+// Close is used to shut down an entire don immediately
+func (c *ccipPlugins) Close() error {
 	if c == nil {
 		return nil
 	}
 	var err error
 
-	// shutdown active commit instance.
-	if c.commit.active != nil {
-		err = multierr.Append(err, c.commit.active.Close())
-	}
-
-	// shutdown candidate commit instance.
-	if c.commit.candidate != nil {
-		err = multierr.Append(err, c.commit.candidate.Close())
-	}
-
-	// shutdown active exec instance.
-	if c.exec.active != nil {
-		err = multierr.Append(err, c.exec.active.Close())
-	}
-
-	// shutdown candidate exec instance.
-	if c.exec.candidate != nil {
-		err = multierr.Append(err, c.exec.candidate.Close())
+	for _, oracle := range *c {
+		err = multierr.Append(err, (*oracle).Close())
```

Comment:
With the above suggestion applied this can be:

```suggestion
		err = multierr.Append(err, oracle.Close())
```